internal/service/cache/cache_redis.go

```diff
@@ -0,0 +1,140 @@
+package cache
+
+import (
+	"bytes"
+	"context"
+	"encoding/gob"
+	"io"
+	"os"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/dskit/cache"
+	"github.com/grafana/dskit/flagext"
+)
+
+type RedisCache[valueType any] struct {
+	client *cache.RedisClient
+}
+
+func newRedisCache[valueType any](cfg RedisConf) (*RedisCache[valueType], error) {
+	client, err := cache.NewRedisClient(
+		//TODO NewLogFmtLogger ? Maybe something else
+		log.NewLogfmtLogger(os.Stdout),
+		"redis-cache",
+		cache.RedisClientConfig{
+			Endpoint:            []string{cfg.Endpoint.String()},
+			Username:            "default",
+			Password:            flagext.SecretWithValue(""),
+			MaxAsyncConcurrency: cfg.MaxAsyncConcurrency,
+			MaxAsyncBufferSize:  cfg.MaxAsyncBufferSize,
+			DB:                  0,
```

Comment:
Why not set DB to cfg.DB here ? Default value would still be 0 and it would be possible to configure